src/smt/smt_engine.h

```diff
@@ -508,7 +510,7 @@ class CVC4_PUBLIC SmtEngine
    *
    * @throw TypeCheckingException, LogicException, UnsafeInterruptException
    */
-  Expr expandDefinitions(const Expr& e);
+  Node expandDefinitions(const Node& e);
```

Comment:
I am surprised that this particular change does not propagate to many more places that are unrelated to abduction. It might be a good idea to separate this particular change (and its propagated changes) to a separate PR. But if it is pretty local to abduction, then I guess it is fine to "merge" these two changes together.

Reply:
It is indeed a simple change, I'd prefer not writing any new Expr-level code, which is why I changed here.